tests/func/test_set_expression.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import print_function, with_statement, absolute_import
+
+from tests.helpers.pattern import ANY
+from tests.helpers.session import DebugSession
+from tests.helpers.timeline import Event
+
+
+def test_set_expression(pyfile, run_as, start_method):
+
+    @pyfile
+    def code_to_debug():
+        import backchannel
+        from dbgimporter import import_and_enable_debugger
+        import_and_enable_debugger()
+        import ptvsd
+        a = 1
+        ptvsd.break_into_debugger()
+        backchannel.write_json(a)
+
+    with DebugSession() as session:
+        session.initialize(
+            target=(run_as, code_to_debug),
+            start_method=start_method,
+            ignore_unobserved=[Event('continued')],
+            use_backchannel=True,
+        )
+        session.start_debugging()
+        hit = session.wait_for_thread_stopped()
+
+        resp_scopes = session.send_request('scopes', arguments={
+            'frameId': hit.frame_id
+        }).wait_for_response()
+        scopes = resp_scopes.body['scopes']
+        assert len(scopes) > 0
+
+        resp_variables = session.send_request('variables', arguments={
+            'variablesReference': scopes[0]['variablesReference']
+        }).wait_for_response()
+        variables = list(v for v in resp_variables.body['variables'] if v['name'] == 'a')
+        assert len(variables) == 1
```

Comment:
You can combine these two in a single assert by asserting directly against the list, i.e. `assert variables = [{...}]`.

Reply:
Done